File: 0x-bonus/yegge/odd_numbers/oddNumbers.js
// Write function to print the odd numbers from 1 to 99.

// const printOddNumbers = (upperBound=99) => {
//   const odd = (num) => num % 2 !== 0;
//
//   for (let n = 1; n <= upperBound; n++) {
//     if (odd(n)) {
//       console.log( n );
//     }
//   }
// };

const printOddNumbers = (upperBound=99) => {
  for (let n = 1; n <= upperBound; n += 2) {
    console.log( n );
  }
};

printOddNumbers();

File: 0x-bonus/yegge/odd_numbers/odd_numbers.rb
def print_odd_numbers(upper=99)

  # (2..99).each do |n|
  #   puts n if n.odd?
  # end

  # puts 3.step(99, 2).to_a.join("\n")

  # puts (1..99).to_a.select(&:odd).join("\n")

  3.step(99, 2).each { |n| puts n }

end

print_odd_numbers()

def print_odd_numbers_r(bound=99, n=3)
  if n <= bound
    puts n
    print_odd_numbers_r(bound, n+2)
  end
end

# print_odd_numbers_r()
